src/Engine/Army.java

```diff
@@ -343,6 +360,24 @@ public GameEventQueue receiveMassDamageEvent(Commander attacker, Map<Unit, Integ
           }
         }
           break;
+        case Persistent:
+        {
+          // As persistent tags is meant to be mostly a sidegrade, give half charge to the primary, and split the rest among the rest.
+          final int primaryCharge = cos[0].calculateMassDamageCharge(minion, damageEntry.getValue());
+          if( cos.length == 1 )
+            cos[0].modifyAbilityPower(primaryCharge);
+          else
+          {
+            cos[0].modifyAbilityPower(primaryCharge / 2);
+            final double tagMultiplier = 0.5 / (cos.length - 1);
+            for( int i = 1; i < cos.length; ++i )
+            {
+              final int tagCharge = cos[i].calculateMassDamageCharge(minion, damageEntry.getValue());
+              cos[i].modifyAbilityPower((int) (tagCharge / tagMultiplier));
```

Comment:
Should this be `tagCharge * tagMultiplier` to match line 316? Also, can we put the logic in one place so we don't have this problem?

Reply:
Good catch.